validation/src/main/scala/hmda/validation/rules/lar/validity/V300.scala

```diff
@@ -0,0 +1,66 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.census.CBSATractLookup
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.dsl.Result
+import hmda.validation.rules.EditCheck
+import hmda.validation.dsl.PredicateCommon._
+import hmda.validation.dsl.PredicateSyntax._
+import hmda.validation.dsl.PredicateRegEx._
+
+object V300 extends EditCheck[LoanApplicationRegister] {
+
+  val cbsaTracts = CBSATractLookup.values
+
+  val validCombination = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }
+
+  val validMdCombination = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }
+
+  val validStateCountyCombination = cbsaTracts.map { cbsa =>
+    (cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }
```

Comment:
Strangely, a quick search didn't find anything handy. So I'll take a shot, with some (over-?)simplification.

A Scala List is basically a linked list. (What I'm about to say would also apply to an array.)  So when you're doing a `contains` in it, you end up traversing the whole list (in the `false` case) or on average half the list (in the `true` case).  In other words, it takes linear time.

A Scala Set, on the other hand, is basically* a hash table. The idea there is that it takes every element and does math (it's really cool, and worth reading about) so that lookup can happen in (effectively, usually) constant time.  For a large number of elements, it's usually a win.  Yes, "large" is a vague term!

*not exactly; for more info go [here](http://docs.scala-lang.org/overviews/collections/sets#sorted-sets) and scroll up a couple of paragraphs.